src/libphonenumber/PhoneNumber.php

```diff
@@ -546,13 +546,17 @@ public function unserialize($serialized)
     {
         $data = unserialize($serialized);
 
-        $this->setCountryCode($data[0]);
-        $this->setNationalNumber($data[1]);
-        $this->setExtension($data[2]);
-        $this->setItalianLeadingZero($data[3]);
-        $this->setNumberOfLeadingZeros($data[4]);
-        $this->setRawInput($data[5]);
-        $this->setCountryCodeSource($data[6]);
-        $this->setPreferredDomesticCarrierCode($data[7]);
+        list (
+            $this->countryCode,
+            $this->nationalNumber,
+            $this->extension,
+            $this->italianLeadingZero,
+            $this->numberOfLeadingZeros,
+            $this->rawInput,
+            $this->countryCodeSource,
+            $this->preferredDomesticCarrierCode
+        ) = $data;
```

Comment:
I had to update `unserialize()` as well, as it could call setters with `null` values, which would then be wrongly typecasted to zeros or empty strings.